lib/rubocop/cop/lint/numeric_operation_with_constant_result.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Certain numeric operations have a constant result, usually 0 or 1.
+      # Subtracting a number from itself or multiplying it by 0 will always return 0.
+      # Additionally, a variable modulo 0 or itself will always return 0.
+      # Dividing a number by itself or raising it to the power of 0 will always return 1.
+      # As such, they can be replaced with that result.
+      # These are probably leftover from debugging, or are mistakes.
+      # Other numeric operations that are similarly leftover from debugging or mistakes
+      # are handled by Lint/UselessNumericOperation.
```

Comment:
There's also `UselessNumericOperation`. Probably we should update its docs as well, as it's kind of similar to this one. Too many cops! 😆